App Data Retriever and Search Term Explorer Merge into Develop

sauyon opened the pull request.

AdamSlack committed:
Started the retriever, need to log to DB

AdamSlack committed:
Started the retriever, need to log to DB

AdamSlack committed:
Started the retriever, need to log to DB

AdamSlack committed:
Started the retriever, need to log to DB

AdamSlack committed:
Added some ToDo's

AdamSlack committed:
Added some ToDo's

AdamSlack committed:
Added some ToDo's

AdamSlack committed:
added search terms table

AdamSlack committed:
added search terms table

AdamSlack committed:
Merge branch 'AppDataRetriever' of github.com:sociam/xray into AppDataRetriever

AdamSlack committed:
started adding functionality to db.js

AdamSlack committed:
Merge branch 'AppDataRetriever' of github.com:sociam/xray into AppDataRetriever

AdamSlack committed:
db.js updated to have search_term table interaction, scrapeSuggested renamed retriever and updated to log to DB, retriever updated to log to db, it still needs to fetch app permissions

AdamSlack committed:
Moved some code around, also added an import function to explorer to add existing files of search terms

AdamSlack committed:
typo in the db query function for retching scraped data

AdamSlack committed:
verified db functionality for explorer.js

AdamSlack committed:
resolved retriever bug not loggin to DB, temporarily set insert playstore app in db.js to have 'f' as the isDownloaded col each time...

AdamSlack commented:
got 2 and a half issues to resolve first :haircut_man:

sauyon commented on pipeline/scraper/db.js:
```diff
@@ -45,6 +45,44 @@ async function insertDev(dev) {
+     */
+    get_search_terms: async() => {
+        logger.debug('Fetching Search Terms');
+        var res = await query('SELECT search_term FROM search_terms WHERE current_date > last_searched + interval \'1 month\'');
```
Might use `...WHERE age(last_searched) > interval \'1 month\'` instead.

sauyon commented on pipeline/scraper/db.js:
```diff
@@ -45,6 +45,44 @@ async function insertDev(dev) {
+    /**
+     *  Query the search_terms table to get a list of terms that are stale
+     */
+    get_search_terms: async() => {
```
Please rename to `getSearchTerms`

sauyon replied:
@AdamSlack would prefer you reply in this thread :rofl: anyway yeah snake casing is the best :+1: but everybody seems to like camel casing here for some reason.

sauyon reviewed (commented):
I think I solved one of them :D

sauyon commented:
But yeah I'll look at this later then.

francisAndrew38 committed:
Update db.js

AdamSlack commented:
Sounds good! That should resolve the issue with current_date not being liked anymore for postgres

AdamSlack commented:
Also, will go through and camel case over underscore all var names. Dunno why I swapped, just force of habit I guess :L

AdamSlack committed:
updated db.js - also resolved issue regarding 'current_date' throwing promise exception.

AdamSlack committed:
Merge branch 'AppDataRetriever' of github.com:sociam/xray into AppDataRetriever

AdamSlack committed:
removed redundant log that also caused logging of 'undefined'

AdamSlack commented:
Should be ready to merge now, pls review! :+1: :+1:

AdamSlack committed:
added debug message

francisAndrew38 committed:
Update db.js

francisAndrew38 committed:
Update db.js

AdamSlack committed:
changed all _ to camelCase

AdamSlack committed:
attempts to fix db inserts

deanott committed:
Merge branch 'develop' into AppDataRetriever

AdamSlack committed:
the issue with the fix of db inserting is fixed, now inserts to the DB when the app data doesnt already exist.

deanott committed:
db insertion aded err check

deanott committed:
retreiveer refactor to find bug + found version was never set to be date when is undefined

deanott committed:
quick app existance function:

deanott committed:
set to only insert newly retrieved data

sauyon commented on pipeline/scraper/db.js:
```diff
@@ -31,20 +31,92 @@ function connect() {
-    if (res.rowCount > 0) {
-        return res.rows[0].id;
-    }
+    try {
```
remove try catch

sauyon commented on pipeline/scraper/db.js:
```diff
@@ -31,20 +31,92 @@ function connect() {
+    /**
+     *  Query the search_terms table to get a list of terms that are stale
+     */
+    //TODO: Rename to 'getStaleSearchTerms'
```
remove TODO lines

AdamSlack committed:
example config updated

sauyon commented on pipeline/scraper/retriever.js:
```diff
@@ -0,0 +1,87 @@
+// }).then(
+//     (app) => logger.info(app),
+//     (err) => logger.err(err.message)
+// );
```
Trailing newlines please.

AdamSlack replied:
Added EOF new lines

sauyon commented on pipeline/scraper/scrape.js:
```diff
@@ -62,7 +62,8 @@ let waitForPipe = () => {
     }
-}; waitForPipe();
+};
+waitForPipe();
```
I would move this as close to the end as possible

sauyon commented on pipeline/scraper/logger.js:
```diff
@@ -6,18 +6,18 @@ Simply logger to correctly error messages to systemd.
-      NOTICE = 5,
-      INFO = 6,
-      DEBUG = 7;
+    ALERT = 1,
```
should be aligned with const

AdamSlack replied:
blame the linter.

sauyon reviewed (commented):
dammit guys

AdamSlack committed:
config changed to export a class instead of json object

AdamSlack committed:
updated retriever to make use of new db

AdamSlack committed:
gone back on the config options, adding it to the wish list for future changes. it's neither Important or urgent to progress.

AdamSlack committed:
moved comments in db.js

AdamSlack committed:
gplay store icon downloading started

AdamSlack committed:
started scraping of app icons for the app data retrieval

sauyon committed:
Merge branch 'develop' into AppDataRetriever

sauyon committed:
Use a class for the DB module

sauyon commented on pipeline/example_config.json:
```diff
@@ -1,25 +1,38 @@
+    "analyzer": {
+        "db": {
+            "user": "analyzer",
+            "password": ";:&!kQu{o>=F+Mm)lnkTaL{_U@&<$I"
```
goddammit guys

sauyon reviewed (commented):
...

sauyon committed:
use bigint for a few fields. closes #24

sauyon committed:
add lquery to the actual value instead of the promise

sauyon committed:
Remove password from example config.

Yes, the password has been changed.
Dammit @AdamSlack.

sauyon commented on db/init_db.sql:
```diff
@@ -30,19 +30,25 @@ create table playstore_apps(
+
+create table search_terms(
+  search_term            text                              primary key not null,
+  last_searched          DATE                                          not null
```
why is DATE the only thing capitalized in the entire file lol

AdamSlack replied:
lol jk fixed.

sauyon commented on pipeline/scraper/explorer.js:
```diff
@@ -4,12 +4,24 @@
+const fs = require('fs-extra');
+const logger = require('./logger.js');
+const wordStoreLocation = config.datadir + '/suggested_words.txt';
+var database = require('./db.js');
```
why is this the only thing that's var

sauyon commented on pipeline/scraper/explorer.js:
```diff
@@ -36,6 +48,27 @@ function writeScrapedWords(word, location) {
+ * Used returns an array where each line is a search term.
+ * @param {*the location of the file that is to be read} file_location 
+ */
+function openSearchTerms(file_location) {
```
are these two functions necessary now?

AdamSlack replied:
not in use right now, but if we wanted to add external files of search terms. really they should be in a separate file as exports i guess!

sauyon commented on pipeline/scraper/explorer.js:
```diff
@@ -59,27 +92,29 @@ function cartesianProductChars() {
  * @param {*The list of words used to get autocompletes} startingWords 
  */
+// TODO: Store scraped word to the Database not txt
 function scrapeSuggestedWords(startingWords) {
```
I think this should be slightly smarter. I'll open an issue for it.

sauyon commented on pipeline/scraper/logger.js:
```diff
@@ -45,4 +45,4 @@ module.exports = {
         log(DEBUG, arguments);
     }
-};
+};
```
trailing newlines!

AdamSlack replied:
blame VS Code. gotta run through linter with --fix. fml :disappointed:

sauyon commented on pipeline/scraper/scrape.js:
```diff
@@ -345,4 +344,4 @@ function main() {
 //         fs.createReadStream(p.join(wordStash,file)).pipe(parser);
 //     });
-// });
+// });
```
trailing newlines!

sauyon commented on pipeline/scraper/retriever.js:
```diff
@@ -0,0 +1,85 @@
+ * @param {*The app data json that is to be inserted into the databae.} app_data 
+ */
+function insertAppData(app_data) {
+
```
get rid of this newline.

AdamSlack replied:
Yes Sir. Done Sir. :vulcan_salute:

sauyon commented on pipeline/scraper/retriever.js:
```diff
@@ -0,0 +1,85 @@
+function insertAppData(app_data) {
+
+    //Checking version data - correct version to update date
+    if (!app_data.version || app_data.version === 'Varies with device') {
```
maybe we should start tagging all versions with updated date instead?

AdamSlack replied:
Might be wise. Perhaps should be revisited in the future.

sauyon commented on pipeline/scraper/retriever.js:
```diff
@@ -0,0 +1,85 @@
+    return db.insertPlayApp(app_data, region);
+}
+
+function updateSearchedTermDate(searchTerm) {
```
this really doesn't need to be a function - that log should probably go into the db module.

sauyon commented on pipeline/scraper/retriever.js:
```diff
@@ -0,0 +1,85 @@
+function insertAppData(app_data) {
+
+    //Checking version data - correct version to update date
+    if (!app_data.version || app_data.version === 'Varies with device') {
```
Also I think this should probably go into the db module.

AdamSlack replied:
Agreed. Should revisit later after Ret' Exp' and Down' are all up and running in an acceptable state.

sauyon commented on pipeline/scraper/retriever.js:
```diff
@@ -0,0 +1,85 @@
+/**
+ * uses db.js to fetch search terms from the database.
+ */
+async function fetch_search_terms() {
```
...why is this even a function

sauyon reviewed (commented):
What's with the `json_dumps` directory?

Also we should nuke a lot of massive commented out sections that we don't need anymore.

sauyon committed:
Add prefix to multiple lines in logger

Replace newlines with newline + <prefix> when logging

sauyon committed:
Update config.go to accept new config structure

Also:
 * added trailing newline to scrape.js
 * fixed a bug where if a dir didn't exist the analyzer would crash

sauyon committed:
Fix DB module to take strings

AdamSlack committed:
When the search term already exists, the client was never released. damn... forgot to add an else to release the client.

AdamSlack committed:
Updated logger to use regex for global replace

AdamSlack committed:
changed the end async promise to make use of bluebirds Promise.each. Cleaner code that way

AdamSlack committed:
Added client.release to updateLastSearchedDate

Forgot to add client.release when updating Last_searched date in the search_terms table in the DB.

sauyon committed:
dammit guys

AdamSlack commented:
Responded to all Comments, pls review :bowing_man:

AdamSlack reviewed (approved):
gone through and made the proposed changes

deanott committed:
added analyze bool

deanott committed:
Merge branch 'AppDataRetriever' of github.com:sociam/xray into AppDataRetriever
